src/libtriton/engines/symbolic/symbolicEngine.cpp

```diff
@@ -1007,6 +1006,10 @@ namespace triton {
         if (ret.size() == 1) {
           /* Synchronize the concrete state */
           this->architecture->setConcreteMemoryValue(mem, tmp->evaluate());
+          /* See #1002: There may be multiple new symbolic expressions when AST_OPTIMIZATIONS are on */
+          for (auto i = id; i != this->uniqueSymExprId; ++i) {
+            inst.addSymbolicExpression(this->getSymbolicExpression(i));
+          }
```

Comment:
I'm not really ok to add `goto` everywhere but I think we can do an exception here to avoid redundant code. So, i think we can replace those 4 lines above with a `goto ret;`. See the label below.

Reply:
I can write a private function ;)